src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -365,6 +369,20 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
                                     target.is_example() ||
                                     profile.test;
 
+                // If this dependency is only available for certain platforms,
+                // make sure we're only enabling it for that platform.
+                let is_platform_same = conds.iter().any(|cond| {
```

Comment:
I'm actually a little surprised this passes tests, this returns `false` in the case that `conds` has no elements, right? I thought that would be the default would be that `conds` only has entries if there's at least one target specific dependency in play...